### PR TITLE
feat: v2 charts with backwards compatibility

### PR DESCRIPTION
**Pull request by bernielomax**
This PR adds proper v2 chart support to abctl. The main changes:

- Added ChartResolver to handle chart references intelligently:
  - Empty chart+version == latest v2 chart
  - Version only -> picks v1/v2 repo based on version number
  - URLs -> fetches and validates via HTTP
  - Local paths -> uses helm client to validate
  
  This makes chart resolution way more robust and handles all the edge cases properly.

- Added version-aware chart value generation that handles both v1 and v2 charts
  - v1 charts keep existing behavior
  - v2 charts get proper workloadLauncher naming and server WEBAPP_URL
  - Pre-release v2 charts (alpha/beta) still use v1 behavior

- Fixed repo URL selection in FindImagesFromChart to use the right repo based on version
  - v1 charts == `airbytehq.github.io/helm-charts`
  - v2 charts == `airbytehq.github.io/charts`

- Cleaned up chart handling in commands
  - Moved client init before chart resolution (was a TODO)
  - Fixed some nil pointer issues in tests
  - Added proper test coverage for all the edge cases

All the v2 chart stuff should work properly now. Tested with both v1 and v2 charts, including pre-releases.

And tested this locally:

Installing older versions:
```
 go run main.go local install --chart-version=1.7.2
```
Results in:
```
NAME         	NAMESPACE    	REVISION	UPDATED                             	STATUS  	CHART               	APP VERSION
airbyte-abctl	airbyte-abctl	1       	2025-07-29 15:59:29.203126 -0700 PDT	deployed	airbyte-1.7.2       	1.7.2      
ingress-nginx	ingress-nginx	1       	2025-07-29 16:00:27.155134 -0700 PDT	deployed	ingress-nginx-4.13.0	1.13.0 
```
 
 Installing latest:
 ```
 go run main.go local install 
```
Results in:
```
NAME         	NAMESPACE    	REVISION	UPDATED                             	STATUS  	CHART               	APP VERSION
airbyte-abctl	airbyte-abctl	1       	2025-07-29 16:06:12.816057 -0700 PDT	deployed	airbyte-2.0.7       	1.7.1      
ingress-nginx	ingress-nginx	1       	2025-07-29 16:08:37.236055 -0700 PDT	deployed	ingress-nginx-4.13.0	1.13.0
```

### CODE REVIEW

File: internal/helm/chart.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	return semver.Compare(v, "v2.0.0") >= 0
+}
+
+// ResolveChartReference resolves an Airbyte chart reference to its full URL/path and version.
+// For empty chart+version, returns latest v2 chart.
+// For version-only, uses v1/v2 repo based on version number.
+// For URLs and local paths, returns as-is with version from chart metadata.
+func (r *ChartResolver) ResolveChartReference(chart, version string) (string, string, error) {
+	if chart == "" {
```
Comment (colesnodgrass) [inline internal/helm/chart.go:56]: Could negate this, return early, and remove a layer of indentation
```
if chart != "" { 
  return
}
```

File: internal/helm/meta.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return chart.Metadata, nil
+}
+
+// GetMetadataForURL fetches a remote chart archive (.tgz) from the given URL and returns its chart metadata.
+func GetMetadataForURL(url string) (*chart.Metadata, error) {
+	resp, err := http.Get(url)
```
Comment (colesnodgrass) [inline internal/helm/meta.go:24]: Does this have a timeout?  I recall the default http client not ever timing out.

**Review by colesnodgrass — APPROVED** [review]

File: internal/helm/chart.go (hunk 1 above)
Comment (colesnodgrass) [reply internal/helm/chart.go:56]: Re-reading this function.  What am I missing?  It returns a `(string, string, error)` but it doesn't look like all the branches return... i.e. if `chart != ""`, what happens?

Comment (bernielomax) [reply internal/helm/chart.go:56]: I'll update the doc, they are not clear. First is the chart path/location, second is the chart version.

File: internal/helm/meta.go (hunk 2 above)
Comment (bernielomax) [reply internal/helm/meta.go:24]: This would be using the default Go HTTP client which has no timeout, so the behavior should be the same. Do you think we should set one?
